url-updater/src/main/java/com/devonfw/tools/ide/url/updater/AbstractUrlUpdater.java

```diff
@@ -190,58 +201,59 @@ protected boolean isSuccess(HttpResponse<?> response) {
+   * @param tool String of the tool
+   * @param version String of the version
+   * @return {@code true} if update of checksum was successful, {@code false} otherwise.
    */
-  private boolean checkDownloadUrl(String url, UrlVersion urlVersion, OperatingSystem os,
-      SystemArchitecture architecture, String checksum) {
-
-    HttpResponse<?> response = doCheckDownloadViaHeadRequest(url);
-    int statusCode = response.statusCode();
-    boolean success = isSuccess(response);
-    String tool = getToolWithEdition();
-    String version = urlVersion.getName();
-    String contentType = response.headers().firstValue("content-type").orElse("undefined");
-    if (success && contentType.startsWith("text")) {
-      logger.error("For tool {} and version {} the download has an invalid content type {} for URL {}", tool, version,
-          contentType, url);
-      success = false;
-    }
-    if (success) {
-      UrlDownloadFile urlDownloadFile = urlVersion.getOrCreateUrls(os, architecture);
-      UrlChecksum urlChecksum = urlVersion.getOrCreateChecksum(urlDownloadFile.getName());
+  private static boolean isChecksumStillValid(String url, UrlVersion urlVersion, OperatingSystem os,
+      SystemArchitecture architecture, String checksum, String tool, String version) {
+
+    UrlDownloadFile urlDownloadFile = urlVersion.getOrCreateUrls(os, architecture);
+    UrlChecksum urlChecksum = urlVersion.getOrCreateChecksum(urlDownloadFile.getName());
+    urlDownloadFile.addUrl(url);
+    urlChecksum.setChecksum(checksum);
+    String oldChecksum = urlChecksum.getChecksum();
+
+    if ((oldChecksum != null) && !Objects.equal(oldChecksum, checksum)) {
+      logger.error("For tool {} and version {} the mirror URL {} points to a different checksum {} but expected {}.",
+          tool, version, url, checksum, oldChecksum);
+      return false;
+    } else {
       urlDownloadFile.addUrl(url);
       urlChecksum.setChecksum(checksum);
-      String oldChecksum = urlChecksum.getChecksum();
-
-      if ((oldChecksum != null) && !Objects.equal(oldChecksum, checksum)) {
-        logger.error("For tool {} and version {} the mirror URL {} points to a different checksum {} but expected {}.",
-            tool, version, url, checksum, oldChecksum);
-        success = false;
-      } else {
-        urlDownloadFile.addUrl(url);
-        urlChecksum.setChecksum(checksum);
-      }
     }
+    return true;
+  }
 
-    doUpdateStatusJson(success, statusCode, urlVersion, url, false);
+  /**
+   * Checks if the content type is valid (not of type text)
+   *
+   * @param url String of the url to check
+   * @param tool String of the tool name
+   * @param version String of the version
+   * @param contentType String of the content type
+   * @return {@code true} if the content type is not of type text, {@code false} otherwise.
+   */
+  private static boolean isContentTypeValid(String url, String tool, String version,
```

Comment:
maybe `isValidDownload` would be more suitable as `text/html` in general is a valid content-type but not for a binary download. Actually we would expect `application/octet-stream` but not every server is 100% compliant so I better went for a blacklist than a whitelist.

Reply:
You could even pass the response as parameter and read the content type within this method, what would make more even sense. Thanks for extracting such smaller methods to make the code better!

Reply:
Maybe even one step further, you could then also do the `isSuccess` check in this method and simplify `checkDownloadUrl` further.

Reply:
I've implemented all of your mentioned points. Thanks for your constructive feedback. I hope that it looks better now.